tee/tee.go

```diff
@@ -79,6 +84,7 @@ type (
 		Recipient common.Address
 		Amount    *big.Int
 		Sig       []byte
+		hash      common.Hash
```

Comment:
you use this as a map key, why does this value also need to be in the struct?

Reply:
It made sense to me since it is tied to the transaction itself and I thought there might be other use-cases. This also allowed me to cache the hash instead of recomputing it every time when its needed But I am okay if we remove this.

Reply:
Ah ok, if you call `Hash()` several times and cache the calls, it's fine.